src/reducers.ts

```diff
@@ -24,49 +23,74 @@ import {
   REMOVE_LOCATION,
   SET_OPEN_CLOSE_UPLOADQUEUE_MODAL,
 } from "./actions";
+import { Contract } from "./types/contractType";
+import { OrganisationWithRoles } from "./types/organisationType";
 
-// todo type this reducer file
-// import {
-//   Contract
-// } from "./types/contractType";
+interface BootstrapState {
+  bootstrap: {
+    user: {
+      id: number,
+      first_name: string,
+      username: string,
+      authenticated: boolean
+    },
+    sso: {
+      login: string,
+      logout: string
+    }
+  } | null,
+  isFetching: boolean,
+  startedFetch: boolean
+}
 
 function bootstrap(
-  state = {
-    bootstrap: {},
+  state: BootstrapState = {
+    bootstrap: null,
     isFetching: false,
     startedFetch: false,
   },
-  action
-) {
+  action: AnyAction
+): BootstrapState {
   switch (action.type) {
     case REQUEST_LIZARD_BOOTSTRAP:
-      return { ...state, isFetching: true, startedFetch: true, };
+      return {
+        ...state,
+        isFetching: true,
+        startedFetch: true
+      };
     case RECEIVE_LIZARD_BOOTSTRAP:
       return {
         ...state,
         bootstrap: action.data,
         isFetching: false,
-        
       };
     default:
       return state;
   }
 }
 
+interface OrganisationState {
+  isFetching: boolean,
+  timesFetched: number,
+  available: OrganisationWithRoles[],
+  selected: OrganisationWithRoles | null
+}
+
 function organisations(
-  state = {
+  state: OrganisationState = {
     isFetching: false,
     timesFetched: 0,
     available: [],
     selected: null,
   },
-  action
-) {
+  action: AnyAction
+): OrganisationState {
+  // I think this part is outdated --> Need to confirm
   // if there is already a selected organisation then this organisation must not have the unique_id field.
   // if it has the unique_id field then empty the selected organisation
-  if (state.selected && state.selected.unique_id !== undefined) {
-    state.selected = null;
-  }
+  // if (state.selected && state.selected.unique_id !== undefined) {
+  //   state.selected = null;
+  // }
```

Comment:
Think this needs to be resolved also, if we leave it like this it'll just stay like that for a long time

Reply:
Removed commented out code